tilings/algorithms/factor.py

```diff
@@ -0,0 +1,280 @@
+                                 if ob.pos[0] in component)
+            requirements = tuple(req for req in self._tiling.requirements
+                                 if req[0].pos[0] in component)
+            factors.append((obstructions, requirements))
+        self._factors_obs_and_reqs = factors
+        return self._factors_obs_and_reqs
+
+    def factorable(self):
+        """
+        Returns `True` if the tiling has more than one factor.
+        """
+        return len(self._get_components()) > 1
+
+    def factors(self):
+        """
+        Returns all the irreducible factors of the tiling.
+        """
+        return [self._tiling.__class__(obstructions=f[0], requirements=f[1],
+                                       minimize=False)
+                for f in self._get_factors_obs_and_reqs()]
+
+    def reducible_factorisations(self):
+        """
+        Iterator over all reducible factorisation that can be obtained by
+        grouping of irreducible factors.
+
+        Each factorisation is a list of tiling.
+
+        For example if T = T1 x T2 x T3 then (T1 x T3) x T2 is a possible non
+        irreducible factorisation.
+        """
+        min_comp = self._get_factors_obs_and_reqs()
+        print(min_comp)
+        for partition in partitions_iterator(min_comp):
+            factors = []
+            for part in partition:
+                obstructions, requirements = zip(*part)
+                factors.append(self._tiling.__class__(
+                    obstructions=chain(*obstructions),
+                    requirements=chain(*requirements),
+                    minimize=False)
+                )
+            yield factors
+
+    def formal_step(self, union=False):
+        """
+        Return a string that describe the operation performed on the tiling.
+        """
+        union_str = 'unions of ' if union else ''
+        return 'The {}factors of the tiling.'.format(union_str)
+
+    @property
+    def constructor(self):
+        """
+        Returns the type of constructor for the factorisation
+        """
+        return 'cartesian'
+
+    def _rule(self, factors, formal_step, workable):
+        """
+        Return the comb_spec_searcher rule for the factorisation where the list
+        of factors is given as a list of tilings.
+
+        TODO: Describe the meaning or workable.
+        """
+        if not self.factorable():
+            return
+        assert isinstance(workable, bool)
+        return Rule(formal_step,
+                    factors,
+                    inferable=[False for _ in factors],
+                    workable=[workable for _ in factors],
+                    possibly_empty=[False for _ in factors],
+                    ignore_parent=workable,
+                    constructor=self.constructor)
+
+    def rule(self, workable=True):
+        return self._rule(self.factors(), self.formal_step(), workable)
+
+    def all_union_rules(self, workable=True):
```

Comment:
By default workable=False in all unions. But work from the maximal factorization.